glusterd2/commands/volumes/volume-smartvol-txn.go

```diff
@@ -48,15 +48,35 @@ func txnPrepareBricks(c transaction.TxnCtx) error {
 			}
 
 			// LV Creation
-			err = deviceutils.CreateLV(b.VgName, b.TpName, b.LvName, b.Size)
-			if err != nil {
-				c.Logger().WithError(err).WithFields(log.Fields{
-					"vg-name": b.VgName,
-					"tp-name": b.TpName,
-					"lv-name": b.LvName,
-					"size":    b.Size,
-				}).Error("lvcreate failed")
-				return err
+			if b.Type == "arbiter" {
+				// All the calculations should be in MB
+				arbiterSize := (4 / 1024) * (b.Size / req.AverageFileSize)
+				// Assigning arbiter brick size to be 20Mb if its lesser than 20Mb
+				if arbiterSize < 20 {
```

Comment:
* is there any specific reason to select `average size` as 20 MB?
in heketi we use `64KB` [heketi-avegare-size](https://github.com/heketi/heketi/blob/63154832abfe16d7d9ebd2f5ded798b0f6fed725/apps/glusterfs/volume_entry.go#L51)

* I think we should not set arbiter size to a default value if its less than 20MB. we should make volume create to fail.

Reply:
20 MB is not average file size here. If the calculated arbiter brick size is less than 20MB, then create brick with 20MB. I think default size of file is not yet considered in this PR

Reply:
if the calculated size is less than `20MB` don't we need to fail it?

Reply:
No. It is like minimum size required to create lv and xfs is 20MB(I think we can increase the default to 100MB). 

Maximum arbiter brick size possible = Size of data brick

Reply:
We have a default value for average file size as 10M. This value is used if the command doesn't have the average file size argument in the creation of replica volume with arbiter brick.
The arbiter brick is always calculated as per this patch. The only change done is, If the arbiter brick size is less than 20M then the size of the brick is itself used as the arbiter brick size.

I haven't failed the create if the arbiter size is calculated to be less than 20M instead I have used it as the size of the brick and passed the creation.

Regarding the average file size, I used 10M as I didn't find any recommended value. As the sizeToMb returns zero for values less than 1M I have changed the average file size to 1M which wont error out and will be the closest to 64K. This is a TODO. Once we change the default value to a unit lesser than M the support for K can be done.